plugin/src/main/kotlin/org/gradle/github/dependencygraph/internal/DependencyExtractor.kt

```diff
@@ -239,8 +231,12 @@ abstract class DependencyExtractor :
     }
 
     private fun writeDependencyGraph() {
-        val builder = GitHubDependencyGraphOutput(gitHubSnapshotParams, getOutputDir())
-        builder.outputDependencyGraph(resolvedConfigurations, buildLayout)
+        createRenderer().outputDependencyGraph(pluginParameters, buildLayout, resolvedConfigurations, getOutputDir())
+    }
+
+    private fun createRenderer(): DependencyGraphRenderer {
+        LOGGER.lifecycle("Constructing renderer: ${getRendererClassName()}")
+        return Class.forName(getRendererClassName()).getDeclaredConstructor().newInstance() as DependencyGraphRenderer
     }
```

Comment:
Consider using the `ServiceLoader` instead?

Reply:
Also, a more useful exception might be helpful, as this will likely be an input people will make a mistake with and it will cause failures they need to debug.

Reply:
Yes it's on the list.

Reply:
@JLLeitschuh I tried (and failed) to get this working using a `ServiceLoader`. I'm not sure if there's something about Kotlin classes that I'm failing to account for, but no services were discovered when I added the appropriate file to `META-INF/services`.

Would you care to take a stab?